src/components/tutorial.jsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Dialog from 'modal-form/dialog';
+import MediaCard from './media-card'
+import { Markdown } from 'markdownz';
+import apiClient from 'panoptes-client/lib/api-client';
+import StepThrough from '../components/step-through';
+
+
+const completedThisSession = {};
+if (window.tutorialsCompletedThisSession) {
+  window.tutorialsCompletedThisSession = completedThisSession;
+}
+
+class Tutorial extends React.Component {
+  componentWillMount() {
+    this.previousActiveElement = document.activeElement;
+  }
+
+  componentWillUnmount() {
+    this.handleUnmount();
+  }
+
+  handleNextClick() {
+    this.stepThrough.goNext();
+  }
+
+  handleUnmount() {
+    if (this.previousActiveElement.focus) {
+      this.previousActiveElement.focus();
+    }
+    const now = new Date().toISOString();
+    completedThisSession[this.props.tutorial.id] = now;
+
+    if (this.props.user) {
+      const projectPreferences = this.props.preferences;
+      if (!projectPreferences.preferences) {
+        projectPreferences.preferences = {};
+      };
+      if (!projectPreferences.preferences.tutorials_completed_at) {
+        projectPreferences.preferences.tutorials_completed_at = {};
+      };
+      const obj = {};
+      projectPreferences.update((
+        obj["preferences.tutorials_completed_at." + this.props.tutorial.id] = now,
+        obj
+      ));
```

Comment:
The convoluted syntax here makes it hard to understand what's being passed to `update`.

I think 
```javascript
projectPreferences.update({`preferences.tutorials_completed_at.${this.props.tutorial.id}`: now});
```
might be easier to read.

Or you could set the hash of changes separately, if line length is an issue.
```javascript
const changes = {
 `preferences.tutorials_completed_at.${this.props.tutorial.id}`: now
};
projectPreferences.update(changes);
```

Reply:
Agreed with @eatyourgreens on the readability here.

Reply:
The backend won't accept that sort of string interpolation =>
`projectPreferences.update({`preferences.tutorials_completed_at.${this.props.tutorial.id}`: now});`

but I'll try the `changes` variable.

Reply:
Could you file that as an issue on the API client?

Reply:
Could you file that as an issue on the API client?

Reply:
Done https://github.com/zooniverse/panoptes-javascript-client/issues/69

Reply:
Thanks! The [json client readme](/zooniverse/json-api-client/blob/master/README.md) specifically says that the syntax for `update` is `person.update({'links.pets': [spot.id]})` so that interpolated key should be supported.